partiql-eval/src/eval.rs

```diff
@@ -182,96 +195,161 @@ pub enum EvalJoinKind {
     Left,
     Right,
     Full,
-    Cross,
-    // TODO revisit JOINS to consider the `Lateral` logic as part of current joins
-    CrossLateral,
 }
 
 #[derive(Debug)]
 pub struct EvalJoin {
     pub kind: EvalJoinKind,
     pub on: Option<Box<dyn EvalExpr>>,
-    pub input_l: Option<Value>,
-    pub input_r: Option<Value>,
+    pub input: Option<Value>,
+    pub left: Box<dyn Evaluable>,
+    pub right: Box<dyn Evaluable>,
```

Comment:
This refactoring made it a lot easier to loop over the left bindings and right bindings (evaluated with respect to each left binding) as [spec 5.6](https://partiql.org/assets/PartiQL-Specification.pdf#subsection.5.6) defines.